src/bank.rs

```diff
@@ -412,6 +412,19 @@ impl Bank {
         Ok(())
     }
 
+    fn program_id_to_string(id: &Pubkey) -> String {
+        let mut map = HashMap::new();
```

Comment:
It would be nice to not receate this hashmap on every call

Reply:
Yea, I agree. Seems the options are use `static mut` with an unsafe block, potentially a crate called phf which is like a perfect hashing library or use an if-else chain. I tried with match but I didn't get it to work. None of the options seemed that good.

Reply:
hmm, phf is nightly-only. lazy_static is another option which I *think* should work on stable.